mountpoint-s3-fs/src/prefetch.rs

```diff
@@ -291,18 +291,28 @@ where
             next_seq_offset = self.next_sequential_read_offset,
             "read"
         );
-        let result = self.try_read(offset, length).await;
-        if result.is_err() {
-            self.reset_prefetch_to_offset(offset);
+
+        match self.try_read(offset, length).await {
```

Comment:
I'm unconvinced this is the right layer to measure cache hits. All data read here was already prefetched in memory, regardless of where it originally came from. For example. multiple reads at the same offset will count as multiple cache misses even if we only made a single request to S3 and kept the data in RAM.